### ISSUE TITLE
Copy constructors on SF objects

### ISSUE THREAD

**Issue by ForeverWintr** [InvestmentSystems/static-frame#188]
I'm writing a generic function that receives a `Frame` (or `FrameGO`), and returns a frame with the same data but additional columns. I want to convert my received Frame to a new FrameGO, but currently I don't think there's an easy way to do that without copying data. 

```python
def add_cols(f: sf.Frame):
    new_f = f.to_frame_go()
```

The above works when `f` is a `Frame`, but fails if it's a `FrameGO`.

An easy solution is to provide an implementation for `FrameGO.to_frame_go`, but I think a more scalable solution is to implement `Frame.from_frame`. I think it can simply be:

```python
    @classmethod
    def from_frame(cls, f) -> "Frame":
        '''
        Copy constructor.
        '''
        # copying blocks does not copy underlying data
        return cls(f._blocks.copy(),
                index=f.index,
                columns=f.columns.values,
                name=f._name,
                own_data=True,
                own_index=True,
                own_columns=False # need to make static only
                )
```

And if implemented as above, I believe `FrameGO.from_frame` (as well as other hypothetical subclasses of `Frame`) will also work. 

And if we're doing that, do we want a `Series.from_series` while we're at it? 

I can submit a pull request that implements this, if you agree.

**Comment by flexatone** [comment]
Many thanks for this issue.

You bring up two things we can improve.

1. `to_frame` and `to_frame_go` should be operational on all containers where it is practical. They are already there on `Series`; `IndexHierarchy` has only `to_frame`.

2.  Instead of `from_frame` methods, I prefer that the default constructor (`__init__`) accept `Frame` instances and do the right thing. This is already the case with `Index` and `IndexHierarchy`, and that approach has been quite useful. We can do the same for `Series`, thought I am not sure how useful that will be.

**Comment by flexatone** [comment]
I have implemented `to_frame` and `to_frame_go` on all relevant containers, and have extended `Series` and `Frame` default constructors (`__init__`) to accept and efficiently use `Series` and `Frame` instances, respectively.

This change will be made available in the 0.6.9 release.
